Automattic/jetpack
# Update Users menu for consistency for Atomic sites.

## Proposed changes:

We want to make Users submenu consistent across Simple, Atomic & Jetpack. This PR is for the Atomic environment.


* Current submenu structure
  * All Users
  * Add New User
  * Profile
  * Subscribers
  * Account Settings

* Desired submenu structure:
  * All Users
  * Add New User
  * **Subscribers** _(moved up for consistency)_
  * **My Profile** _(renamed from "Profile" for consistency)_
  * Account settings

This PR also slips in a small fix for Simple environment where "Add New" is renamed to "Add New User". See related patch: Fixing simple site submenu inconsistencies @ D128198-code


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:







**Preparation**
* Have a WoA site ready (Hint: `<mc_base_url>/atomic/wpcom-dev-blogs`)

**Calypso Menu**
* On your sandbox, run `bin/jetpack-downloader test jetpack fix/atomic-users-menu-consistency`.
* Sandbox `public-api.wordpress.com`.
* Go to `wordpress.com/people/team/your_woa_site.wordpress.com`.
* You should see the Users has the following submenu structure:
  * All Users
  * Add New User
  * Subscribers
  * My Profile
  * Account settings

**Classic Admin Menu**
* On your WoA site, install [Jetpack Beta](https://jetpack.com/download-jetpack-beta/). 
* Go to Jetpack > Beta Tester (or `your_woa_site.wordpress.com/wp-admin/admin.php?page=jetpack-beta&plugin=jetpack`).
  * Under "Search for a Feature Branch", enter `fix/atomic-users-menu-consistency`.
  * Activate this branch.
* Go to `your_woa_site.wordpress.com/wp-admin/user-new.php`.
* You should see the Users has the following submenu structure:
  * All Users
  * Add New User
  * Subscribers
  * My Profile
  * Account settings

**Test the unit test**:
- Run `jetpack docker phpunit -- --filter=Test_Atomic_Admin_Menu`.
  - All tests should pass. One test is skipped because that's for multisite only.
- Run `jetpack docker phpunit-multisite -- --filter=Test_Atomic_Admin_Menu`.
  - All tests should pass.

| Where | Before | After |
| --- | --- | --- | 
| Calypso Menu | <img width="932" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/d11489c7-7593-457b-8d67-8338a60d3527"> | <img width="898" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/8f38c338-ca45-4cfe-a65a-50dfea4910ed"> |
| Classic Admin Menu | <img width="1096" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/f4826688-75ca-4078-8458-24f8ea8607ab"> | <img width="1166" alt="image" src="https://github.com/Automattic/jetpack/assets/1287077/0fed98f8-4f95-4abc-9dc4-f2feef241505"> |
